app/react/V2/Routes/Settings/Thesauri/specs/Thesauri.spec.tsx

```diff
@@ -0,0 +1,469 @@
+import { Provider as JotaiProvider } from 'jotai';
+import { Provider } from 'react-redux';
+import createMockStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+import Immutable from 'immutable';
+import { has } from 'lodash';
+import { atomsGlobalState } from 'V2/shared/testingHelpers';
+import { templatesAtom } from 'app/V2/atoms';
+import { ThesauriList, thesauriLoader } from '../ThesauriList';
+import { EditThesaurus } from '../EditThesaurus';
+import { editThesaurusLoader } from '../helpers';
+import { savedThesaurus, thesauri } from './fixtures';
+
+const deleteFn = jest.fn();
+const saveFn = jest.fn();
+const mockUseLoaderData = jest
+  .fn()
+  .mockImplementation((params, _headers) => (!has(params, '_id') ? thesauri : [savedThesaurus]));
+const mockDeleteThesauri = deleteFn.mockImplementation((_params, _headers) => ({
+  ok: true,
+}));
+const mockSaveThesauri = saveFn.mockImplementation((_params, _headers) => savedThesaurus);
+
+jest.mock('app/V2/api/thesauri', () => ({
+  __esModule: true,
+  default: {
+    ...jest.requireActual('app/V2/api/thesauri').default,
+    getThesauri: (params: { _id: string }, headers?: IncomingHttpHeaders) =>
+      mockUseLoaderData(params, headers),
+    delete: (params: { _id: string }, headers?: IncomingHttpHeaders) =>
+      mockDeleteThesauri(params, headers),
+    save: (params: { _id: string }, headers?: IncomingHttpHeaders) =>
+      mockSaveThesauri(params, headers),
+  },
+}));
+
+describe('Settings Thesauri', () => {
+  beforeEach(() => {
+    jest.spyOn(console, 'error').mockImplementation(jest.fn());
+    jest.spyOn(console, 'warn').mockImplementation(() => {});
+  });
+
+  describe('ThesauriList', () => {
+    let renderResult: RenderResult;
+    const store = atomsGlobalState();
+    store.set(templatesAtom, [
+      {
+        _id: 'template1',
+        name: 'Document',
+        properties: [
+          {
+            _id: 'property1',
+            name: 'prop1',
+            label: 'property1',
+            type: 'select',
+            content: 'thesaurus2',
+          },
+          {
+            _id: 'property2',
+            name: 'prop2',
+            label: 'property2',
+            type: 'select',
+            content: 'newThesaurus1',
+          },
+        ],
+      },
+    ]);
+    const router = createMemoryRouter(
+      [
+        { index: true, path: '/', element: <ThesauriList />, loader: thesauriLoader({}) },
+        { path: '/settings/thesauri/new', element: <EditThesaurus /> },
+        {
+          path: '/settings/thesauri/edit/newThesaurus1',
+          element: <EditThesaurus />,
+          loader: editThesaurusLoader({}),
+        },
+        {
+          path: '/edit/newThesaurus1',
+          element: <EditThesaurus />,
+          loader: editThesaurusLoader({}),
+        },
+      ],
+      {
+        initialEntries: ['/'],
+      }
+    );
+
+    const renderComponent = () => {
+      const reduxStore = createMockStore([thunk])(() => ({
```

Comment:
If we incorporate this method of testing routes, the mock store default basic values should probably be extracted into the shared folder for V2 ``V2/shared`